packages/core/src/types/config/index.ts

```diff
@@ -19,6 +19,35 @@ import type { BaseFields } from './fields';
 import type { ListAccessControl, FieldAccessControl } from './access-control';
 import type { ListHooks } from './hooks';
 
+export type StorageKind =
+  | {
+      kind: 'local';
+      type: 'file' | 'image';
+      storagePath?: string;
+      baseUrl?: string; // e.g. https://blah.com/images
+      removeFileOnDelete?: boolean;
+    }
+  | {
+      kind: 's3';
+      type: 'file' | 'image';
+      signed?: { expiry: number };
+      proxied?: {
+        baseUrl: string;
+        generate?: boolean; // if false, allow proxied requests but return urls directly to s3
+      };
```

Comment:
Is this stuff being implemented in this PR or later?

Reply:
Plan is still this PR - got some base bits and will try and push tonight.